docs/4.4/api-reference.md

```diff
@@ -54,14 +54,131 @@ certificate lifetime.
 
 # gRPC APIs
 
+## Client QuickStart
+
+```go
+{!examples/go-client/client.go!}
+```
+
 ## Audit Events API
 Coming Soon
 
 ## Certificate Generation API
 Coming Soon
 
+## Roles APIs
+
+[Roles](http://localhost:6600/enterprise/ssh-rbac/#roles) are used to define what resources and actions a user is allowed/denied access to.
+
+#### Create a new role
+
+```go
+import github.com/gravitational/teleport/lib/services
+
+role, _ := services.NewRole(roleName, services.RoleSpecV3{
+	Allow: services.RoleConditions{
+    Logins: []string{login},
+	},
+})
+
+client.UpsertRole(ctx, role)
+```
+
+#### retrieve roles
+
+```go
+roles, err := client.GetRoles()
+if err != nil {
```

Comment:
might skip the error handling in this snippet

Reply:
I'm thinking we always include error handling in our code snippets. I imagine a lot of cut/paste will be happening and if we don't handle errors, neither will customers.

Reply:
I agree that doing an error check here feels clunky, how about just `roles, err := client.GetRoles()`, so they either have to handle the error, or manually change it to `_` (which no go developer should do usually).

Reply:
Let's do the right thing - error check is OK